internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -522,7 +521,7 @@ func resourceApiManagementApiRead(d *pluginsdk.ResourceData, meta interface{}) e
 	}
 
 	name := getApiName(id.ApiId)
-	newId := api.NewApiID(id.SubscriptionId, id.ResourceGroupName, id.ServiceName, name)
+	newId := api.NewApiID(id.SubscriptionId, id.ResourceGroupName, id.ServiceName, id.ApiId)
```

Comment:
I think we should keep this as `name` as that will get any revisions that have been specified in the id, otherwise this is no different than what is currently used as `id`

`
newId := api.NewApiID(id.SubscriptionId, id.ResourceGroupName, id.ServiceName, name)
`

Reply:
Hi @mbfrahry thanks for your feedback. The code has been updated. Also renamed the file name of the test data. Could you please take another look?